Myndex2023 1 (#25144)
mdn/content: files/en-us/web/accessibility/understanding_colors_and_luminance/index.md

![On the left is a cone mosaic of standard vision, and on the right is that of someone with protanopia where they are missing the red cones.](conemosaics.jpg)

On the left is the central cone mosaic of standard vision, and on the right is that of someone with protanopia, a form of color vision deficiency, where they are missing the red cones. (Illustration by Mark Fairchild of RIT, [wikicommons](https://commons.wikimedia.org/wiki/file:conemosaics.jpg))

The red and the green cones join together to create luminance, which we can think of as lightness/darkness without regard to hue. Separately, the red, green, and blue cones allow for standard vision to perceive millions of colors. For accessibility, it's important to know that our brain processes luminance separately from color (hue and colorfulness).

Luminance provides for fine vision details, including differentiating edges and text. Hue and colorfulness carry a third of the detail of luminance. Image data compression takes advantage of this fact. As an example, [h.264 video codec](/en-US/docs/Web/Media/Formats/Video_codecs) samples color at a fourth of the resolution of the luminance.

> For accessibility, this means that it is luminance contrast that is critically important for text. Color, as in hue and colorfulness, is important for _distinguishing_ items such as different lines on a map, or bars in a graph.

Author: Just FYI the reason for the **_"it is"_** distinction is that it is luminance contrast that is important for text, and not color contrast. Readability is a function of luminance contrast and not color contrast (hue).

The calculations for evaluating luminance are spelled out in WCAG 2.1 in the definition for [relative luminance](https://www.w3.org/TR/WCAG21/#dfn-relative-luminance), and is as follows:

R, G, and B have distinct weights in the relative luminance formula.For the sRGB colorspace, the relative luminance of a color is defined as `L = 0.2126 * R + 0.7152 * G + 0.0722 * B` where red, green and blue, _R_, _G_ and _B_, are defined as:

- if `RsRGB <= 0.04045` then `R = RsRGB/12.92` else `R = ((RsRGB+0.055)/1.055) ^ 2.4`
- if `GsRGB <= 0.04045` then `G = GsRGB/12.92` else `G = ((GsRGB+0.055)/1.055) ^ 2.4`
- if `BsRGB <= 0.04045` then `B = BsRGB/12.92` else `B = ((BsRGB+0.055)/1.055) ^ 2.4`

These formulas come from [WCAG&nbsp;2.1](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html#dfn-relative-luminance) and [sRGB (IEC 61966-2-1:1999)](https://webstore.iec.ch/publication/6169). _(NOTE 0.04045 is the official IEC standard. WCAG&nbsp;2 was originally drafted with an older value 0.03928. In May&nbsp;2021, this was corrected to 0.04045 in the WCAG&nbsp;2.1 document)._

> Calculating relative luminance is not simple, but there are plenty of tools that can do it for you. Here are a few which show the relative luminance along with the WCAG&nbsp;2 contrast ratio.

Reviewer: I think we should actually get rid of this whole block. It is beyond the scope of an introductory article about color and luminance.

Author: Yes I agree, if your intent is to have a plain language thing having something like this is out of scope, but I didn't want to delete anything because I wasn't quite certain

Reviewer: COol. This was removed in a PR that I had submitted before seeing this PR, so there will be a conflict. I don't think that conflict will show until this PR is approved, so I'll resolve the conflict then.

Reviewer: NVM. I was able to resolve.

- [Lea Verou's contrast tool calculates relative luminance](https://contrast-ratio.com)
- [PlanetCalc Relative luminance calculation](https://planetcalc.com/7779/)
- [Relative luminance calculator at Topster](https://www.topster.net/relative-luminance/)

## Perceiving color

> Color is our perception of the narrow band of visible light, from red through yellow and green to blue. Our sensitivity to these various hues of color are not equal. The light sensitive cells in our [eyes](https://www.verywellhealth.com/eye-cones-5088699), called cones, are tuned to perceive some colors more than others. About 65% of cones are _most_ sensitive to a yellow/green, but also respond to red (we'll call these "red" cones). 30% are green sensitive, and only [5% are blue sensitive](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0144891#sec001). While there are far fewer blue-sensitive cones than the other two types, these cones are very sensitive, which partially makes up for their smaller numbers.

Reviewer: The original numbers are from https://www.verywellhealth.com/eye-cones-5088699. The updates to the percentages contradict the linked article.  Alos, I don't see the 65/30/5 in the journals.plos.org link.

Author: Values are in the listed cite in one of the tables. 

I have other authoritative sources as well (below). The cite listed by www.verywellhealth.com goes to AAO.org, and while you might think it's authoritative, it doesn't list any actual references or cites of its own, and there's no indication as to who wrote it. As such I don't know if it was something just cobbled together by a web designer, because there's no citable references there (this is a major problem on the Internet).

(Ahnelt et al., 1987) found a small ring immediately around the fovea that was 10%-15%, but 7% overall, that was a bit of an outlier. More recent research has it at 5%. Do keep in mind that there are individual differences, so when I say 5% it's really between about 4.5% to 5.5% or so, and as high as my 6.5% for somebody who has protanopia. There are some older studies that do indicate more, I try to keep my references to the most recent 10 to 20 years.

### _Additional references:_

**From** [Structure of cone photoreceptors](https://www.blueconemonochromacy.org/wp-content/uploads/2011/03/Cone-Structure-Review-2k9-2.pdf)
> _the spatial distribution of S cones across the retina differs from that of L and M cones in several respects. S cones constitute only **about 5%** of the total cone population (Roorda et al., 2001), they are more peripherally located in the retina and are absent from the center of the human fovea._

Link only: the [Roorda paper from 2001](https://www.sciencedirect.com/science/article/pii/S0042698901000438?via%3Dihub)

**From** [S-cone dystrophy in tritan](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2082754/) 
> _S cones are sparse in the normal retina, comprising about 5% of the photoreceptors, with their highest density (>2000 cones/mm2) at about 1.0-deg[a](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2082754/#FN1) eccentricity[35](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2082754/#R35), but with large variations between individuals.[36](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2082754/#R36)_

Please let me know if you'd like more references.

Author: To be clear, **S** cones are "blue" **M** are "green" and **L** are "red".

**_Sidebar: I didn't include information below, because I felt it was too much for plain language_**

"Red Green blue" are not the correct names for the cones, but it's not uncommon for a plain language document to use them, even though strictly speaking it's not correct.

The L cone, which here is being referred to as _red_, actually has a peak response at **_yellowgreen_**. It gained the name "red" because it _extends down_ from green _into red_, while the M cone has a peak response at a deeper green, it does **_not_** extend down into deep red. (I.e. **both L & M cones have peak response in green**, and this is largely why the majority of luminance is green. 

For accessibility purposes, it's why most pure greens should "usually" be the lightest of two colors while pure reds and blues should generally be the darkest (if any kind of detail is a factor).

The M cone is essentially a genetic mutation that exists in primates, and M sensitivity overlaps almost completely with the L cone, but that very small differential makes it easier for us to pick out ripe oranges in a tree...

I'm sure most of this minutia is outside the needs of this document....

> Deep, pure blue is perceived differently than other colors for multiple reasons. First, because the blue cones do not contribute to luminance; second, we have far fewer blue cones than red or green; and third, the blue-sensitive cones are scattered semi-evenly throughout the peripheral-vision area, while most of the red and green cones are densely packed in the narrow central vision ([fovea centralis](https://en.wikipedia.org/wiki/Fovea_centralis)). In fact, at the very center of vision, less than about third of a degree of visual angle, there are [no blue cones](https://www.cis.rit.edu/people/faculty/montag/vandplite/pages/chap_9/ch9p1.html) at all.

Reviewer: ```suggestion
Deep, pure blue is perceived differently than other colors as blue cones do not contribute to luminance and we have far fewer blue cones than red or green.
```

Author: **_Yes._** As I said I was trying to keep intact what was already there and only make updates, so I didn't want to be coming in and doing wholesale deletions.

> ![On the left is a cone mosaic of standard vision, and on the right is that of someone with protanopia where they are missing the red cones.](conemosaics.jpg)

Reviewer: Developer tools provide these tools, enabling blurred vision and the more common types of color vision deficiency simulations. this is likely more useful for developing empathy for protanopia and the ability to develop with color and luminance in mind..

Author: Posted here first because it was the area I was working on, but when you take a look at the image which I just got fixed and hopefully it's showing now, you'll see that it clearly displays the density of the three different cone types. It should be clear looking at this why deep blue has such poor resolution and why it should never be used for detail (against Black).

On the left is the central cone mosaic of standard vision, and on the right is that of someone with protanopia, a form of color vision deficiency, where they are missing the red cones. (Illustration by Mark Fairchild of RIT, [wikicommons](https://commons.wikimedia.org/wiki/file:conemosaics.jpg))

The red and the green cones join together to create luminance, which we can think of as lightness/darkness without regard to hue. Separately, the red, green, and blue cones allow for standard vision to perceive millions of colors. For accessibility, it's important to know that our brain processes luminance separately from color (hue and colorfulness).

> Luminance carries all the fine details in an image, as well as edges, and particularly text. Hue and colorfulness carries a third of the detail of luminance. Image data compression takes advantage of this fact. As an example, the popular h.264 video codec samples color at a fourth the resolution of the luminance.

Reviewer: ```suggestion
Luminance carries all the fine details in an image, as well as edges, and particularly text. Hue and colorfulness carries a third of the detail of luminance. Image data compression takes advantage of this fact. As an example, the popular h.264 [video codec](/en-US/docs/Web/Media/Formats/Video_codecs) samples color at a fourth the resolution of the luminance.
```

Reviewer: Readers may think "images" refers to only images and not 'the image of the screen', so the first half of the edit is an attempt to remove the word "images", but may not be exactly what you intended.

Author: What you wrote is clearer, thank you I agree. This is another where I was trying to stick with what was already there.

For accessibility, this means that it is luminance contrast that is critically important for text. Color, as in hue and colorfulness, is important for _distinguishing_ items such as different lines on a map, or bars in a graph.

> A convenient way to think of this is in the case of old-school broadcast television, the luminance is the "black &amp; white" signal, which both the black &amp; white and color television sets received. Those old color televisions also received a [small additional signal](https://en.wikipedia.org/wiki/YIQ) of _just_ color information. The small additional signal was subtracted from the luminance for the green primary, and added to luminance for the red primary, and finally partly added &amp; partly subtracted for the blue primary.

Reviewer: ```suggestion
```

Author: So was attempting to keep in the reference to television signals although I don't know that that was strictly necessary, this is how I sometimes explain luminance vs color, of course it only work for those of us who had to deal with coat hanger wire and aluminum foil to watch television LOL. Certainly if you don't feel it's adding to the conversation or it's confusing drop it...

> Another essential point to consider is the color or luminance that is surrounding a color. The color will appear differently depending on what is surrounding it. In the following image both the yellow dots and the grey of the square they are on, are identical in terms of the sRGB color value that is being displayed on your monitor. Your context-sensitive perception of these colors makes them appear so different, as your brain's image processing adjusts the perception based on what it thinks is in shadow or not.

Reviewer: ```suggestion
Another essential point to consider is the color or luminance that is surrounding a color. The color will appear differently depending on what is surrounding it. In the following image both the yellow dots and the grey of the square they are on, are identical in terms of the sRGB color values displayed. Your context-sensitive perception of these colors makes them appear different as your brain's image processing adjusts the perception based on what it thinks is in shadow or not.
```

Author: Yes
